packages/rum-core/src/domain/rumEventsCollection/action/listenActionEvents.ts

```diff
@@ -37,7 +38,15 @@ export function listenActionEvents({ onClick }: { onClick(context: OnClickContex
       (clickEvent: MouseEvent) => {
         if (clickEvent.target instanceof Element) {
           // Use a scoped variable to make sure the value is not changed by other clicks
-          const userActivity = { selection: hasSelectionChanged }
+          const userActivity = {
+            selection: hasSelectionChanged,
+            input: hasInputChanged,
+          }
+          if (!hasInputChanged) {
+            setTimeout(() => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about wrapping the timeout callback in a monitor?